src/publishers/s3.js

```diff
@@ -0,0 +1,68 @@
+import AWS from 'aws-sdk';
+import debug from 'debug';
+import path from 'path';
+import s3 from 's3';
+
+import asyncOra from '../util/ora-handler';
+
+const d = debug('electron-forge:publish:s3');
+
+AWS.util.update(AWS.S3.prototype, {
+  addExpect100Continue: function addExpect100Continue() {
+    // Hack around large upload issue: https://github.com/andrewrk/node-s3-client/issues/74
+  },
+});
+
+export default async (artifacts, packageJSON, forgeConfig, authToken, tag) => {
+  const s3Config = forgeConfig.s3 || {};
+  s3Config.secret = authToken || process.env.S3_SECRET;
```

Comment:
Can we use the names that all of the AWS SDKs use for access/secret keys?

http://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-environment.html